pkg/acceptance/helpers/common.go

```diff
@@ -19,3 +25,59 @@ func EnsureQuotedIdentifiersIgnoreCaseIsSetToFalse(client *sdk.Client, ctx conte
 	}
 	return nil
 }
+
+func EnsureScimProvisionerRolesExist(client *sdk.Client, ctx context.Context) error {
+	log.Printf("[DEBUG] Making sure Scim Provisioner roles exist")
+	roleIDs := []sdk.AccountObjectIdentifier{snowflakeroles.GenericScimProvisioner, snowflakeroles.OktaProvisioner}
+	currentRoleID, err := client.ContextFunctions.CurrentRole(ctx)
+	if err != nil {
+		return err
+	}
+	for _, roleID := range roleIDs {
+		_, err := client.Roles.ShowByID(ctx, roleID)
+		if err != nil {
+			return err
+		}
+		grants, err := client.Grants.Show(ctx, &sdk.ShowGrantOptions{
+			Of: &sdk.ShowGrantsOf{
+				Role: roleID,
+			},
+		})
+		if err != nil {
+			return err
+		}
+		if !hasGranteeName(grants, currentRoleID) {
```

Comment:
Should we also check for those roles to be granted to the current user? Otherwise, we won't be able to call `use role` on them, was it intentional or not that useful to do for those roles?

Reply:
I think this check is sufficient - we don't call `use role` for these roles.